test/e2e/configurable_route_test.go

```diff
@@ -39,12 +39,12 @@ func TestConfigurableRouteRBAC(t *testing.T) {
 			t.Fatalf("failed to get ingress resource: %v", err)
 		}
 		ingress.Spec.ComponentRoutes = nil
-		if err := kclient.Update(context.TODO(), ingress); err != nil {
-			t.Errorf("failed to restore cluster ingress.spec resource to original state: %v", err)
+		if err := eventuallyUpdateIngressSpec(t, ingress.Spec); err != nil {
+			t.Fatalf("failed to restore cluster ingress spec resource to original state: %v", err)
```

Comment:
This is the cleanup handler; does it make sense for a failure to revert the spec to prevent reverting the status?

Reply:
It's not always a failure though. I'm attempting to deflake the test. If we want to make more wholesale changes than that warrants a different PR.

Reply:
@Miciah I changed the calls from `Fatalf` back to `Errorf`.